### PR TITLE
Add example demonstrating how to enable / disable diagnostics

### LINKED ISSUE #14569: Document how to toggle individual `bevy_diagnostic` plugins at runtime
## What problem does this solve or what need does it fill?

It's hard for users to discover that diagnostics plugins can be toggled at runtime.

## What solution would you like?

In the docs for diagnostics plugins (and the plugin for logging them), link to and/or explain how they can be toggled via `DiagnosticsStore`.

### PR DESCRIPTION
**Pull request by Nihilistas**
# Objective

fixes #14569

## Solution

added an example to the diagnostic examples and linked the code to the docs of the diagnostic library itself.

## Testing

I tested locally on my laptop in a web browser. Looked fine. You are able to collapse the whole "intro" part of the doc to get to the links sooner (for those who may think that including the example code here is annoying to scroll through)

I would like people to run ```cargo doc```  and go the bevy_diagnostic page to see if they have any issues or suggestions.

---

## Showcase

Screenshot 2024-08-14 at 12 52 16

### CODE REVIEW

**Comment by BD103** [comment]
Is there a specific reason you created a separate markdown file for the example, instead of just including it directly into `lib.rs`?

**Comment by Nihilistas** [comment]
I assumed there may be further examples of usage in the future, so I made
it file-per-example to not bloat the lib.rs

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
I prefer the simpler, more local organization. We can split it out later if we run into problems. Docs look good though, so once this is moved into lib.rs this has my approval.

File: Cargo.toml [hunk 1]
```diff
@@ -3395,3 +3406,156 @@ name = "Monitor info"
 description = "Displays information about available monitors (displays)."
 category = "Window"
 wasm = false
+
+# Copy this file to `config.toml` to speed up your builds.
```
Comment (alice-i-cecile) [inline Cargo.toml:3410]: This shouldn't be here :)

**Comment by alice-i-cecile** [comment]
Sure, an example works well too 👍🏽

File: Cargo.toml (hunk 1 above)
Comment (Nihilistas) [reply Cargo.toml]: my bad

File: crates/bevy_diagnostic/src/lib.rs [hunk 2]
```diff
@@ -7,9 +7,15 @@
     html_favicon_url = "https://bevyengine.org/assets/icon.png"
 )]
 
-//! This crate provides a straightforward solution for integrating diagnostics in the [Bevy game engine](https://bevyengine.org/).
-//! It allows users to easily add diagnostic functionality to their Bevy applications, enhancing
-//! their ability to monitor and optimize their game's.
+//!This crate provides a straightforward solution for integrating diagnostics in the [Bevy game engine](https://bevyengine.org/).
```
Comment (alice-i-cecile) [inline crates/bevy_diagnostic/src/lib.rs:10]: IMO we should leave this file unchanged: the example is enough.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Sorry for nitpicking on organization 😅

**Comment by Nihilistas** [comment]
> Sorry for nitpicking on organization 😅

all good.

**Review by alice-i-cecile — APPROVED** [review]

**Review by BD103 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**